.github/workflows/unified-dogfooding.yml

```
@@ -0,0 +1,33 @@
+name: Unified Dogfooding scans
+on:
+  schedule:
+    # Run the workflow every day at 04:00 UTC
+    - cron: '0 4 * * *'
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - master
+jobs:
+  build:
+    runs-on: ubuntu-24.04-large
+    name: Build and run shadow scans
+    permissions:
+      id-token: write
+      contents: write
+    steps:
+      - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+      - uses: SonarSource/ci-github-actions/build-gradle@master # dogfood
+        with:
+          run-shadow-scans: true
+          artifactory-deploy-repo: sonarsource-public-qa
+          public: false
```

Please don't use the public property and set the necessary properties directly.
```suggestion
          artifactory-reader-role: private-reader
          artifactory-deployer-role: qa-deployer
```